### PR TITLE
feat: added api connectivity info in diagnostics

### PR DESCRIPTION
**Pull request by AdiGajbhiye**
## Overview

- checking if extension has connectivity with the backend by making an unauthorized api call

Screenshot 2024-08-01 at 6 26 03 PM

Screenshot 2024-08-01 at 6 25 07 PM

### Problem

Describe the problem you are solving. Mention the ticket/issue if applicable.

### Solution

Describe the implemented solution. Add external references if needed.

### Screenshot/Demo

A picture is worth a thousand words. Please highlight the changes if applicable.

### How to test

- Steps to be followed to verify the solution or code changes
- Mention if there is any settings configuration added/changed/deleted

## Checklist

- [ ] I have run this code and it appears to resolve the stated issue
- [ ] `README.md` updated and added information about my change

### CODE REVIEW

**Comment by anandgupta42** [comment]
@AdiGajbhiye  if the user is not able to connect to the backend we should fail fast and give them a relevant error message.

**Review by saravmajestic — APPROVED** [review]
Is there any corresponding backend change to be deployed?

File: src/altimate.ts [hunk 1]
```diff
@@ -746,6 +746,24 @@ export class AltimateRequest {
...
   }
 
+  async checkApiConnectivity() {
+    const url = `${AltimateRequest.ALTIMATE_URL}/health`;
+    try {
+      const response = await fetch(url, { method: "GET" });
+      const { status } = (await response.json()) as { status: string };
+      return status === "ok";
+    } catch (e) {
+      this.dbtTerminal.error(
```
Comment (mdesmet) [inline src/altimate.ts:756]: Please include the exception in the diagnostics. This might be helpful

**Review by mdesmet — COMMENTED** [review]
LGTM

**Pull request merged**